tests/ui/self/dyn-dispatch-do-not-mix-normalized-and-unnormalized.rs

```diff
@@ -0,0 +1,26 @@
+//@ check-pass
+
+// Regression test for <https://github.com/rust-lang/rust/issues/138937>.
+
+// Previously, we'd take the normalized param env's clauses which included
+// `<PF as TraitC>::Value = i32`, which comes from the supertraits of `TraitD`
+// after normalizing `<PF as TraitC>::Value = <PF as TraitD>::Scalar`. Since
+// `normalize_param_env_or_error` ends up re-elaborating `PF: TraitD`, we'd
+// end up with both versions of this predicate (normalized and unnormalized).
+// Since these projections preds are not equal, we'd fail with ambiguity.
```

Comment:
its *during* `normalize_param_env_or_error` that we error due to there being two candidates?

Reply:
We error during `normalize_param_env_or_error` b/c we have two candidates for normalizing <PF as TraitC>::Value`, which shows up in the supertrait `TraitB<Self::Value>`. The first candidate comes from normalizing the param-env one time, and the second comes from the second time we normalize it.